pkg/importer/s3-datasource.go

```diff
@@ -128,20 +127,55 @@ func (sd *S3DataSource) Close() error {
 
 func createS3Reader(ep *url.URL, accessKey, secKey string) (io.ReadCloser, error) {
 	klog.V(3).Infoln("Using S3 client to get data")
-	bucket := ep.Host
-	object := strings.Trim(ep.Path, "/")
-	mc, err := newClientFunc(accessKey, secKey, false)
+
+	endpoint := ep.Host
+	klog.Infof("Endpoint %s", endpoint)
+
+	bucket, object := extractBucketAndObject(ep.Path)
+
+	klog.V(1).Infof("bucket %s", bucket)
+	klog.V(1).Infof("object %s", object)
+	svc, err := newClientFunc(endpoint, accessKey, secKey)
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not build minio client for %q", ep.Host)
+		return nil, errors.Wrapf(err, "could not build s3 client for %q", ep.Host)
+	}
+
+	objInput := &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(object),
 	}
-	klog.V(2).Infof("Attempting to get object %q via S3 client\n", ep.String())
-	objectReader, err := mc.GetObject(bucket, object, minio.GetObjectOptions{})
+	objOutput, err := svc.GetObject(objInput)
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not get s3 object: \"%s/%s\"", bucket, object)
 	}
+	objectReader := objOutput.Body
 	return objectReader, nil
 }
 
-func getS3Client(accessKey, secKey string, secure bool) (S3Client, error) {
-	return minio.NewV4(common.ImporterS3Host, accessKey, secKey, secure)
+func getS3Client(endpoint, accessKey, secKey string) (S3Client, error) {
+	creds := credentials.NewStaticCredentials(accessKey, secKey, "")
+	region := extractRegion(endpoint)
+	sess, err := session.NewSession(&aws.Config{
+		Region:           aws.String(region),
+		Endpoint:         aws.String(endpoint),
+		Credentials:      creds,
+		S3ForcePathStyle: aws.Bool(true),
+	},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	svc := s3.New(sess)
+	return svc, nil
+}
+
+func extractRegion(s string) string {
+	splitted := strings.Split(s, ".")
+	return splitted[0]
+}
+
+func extractBucketAndObject(s string) (string, string) {
+	bucket, object := filepath.Split(s)
```

Comment:
I am sorry for confusing you, filepath.Split is not what we want, it doesn't split on the separator it splits a path into a path and file part. But we want the first part of the path and the rest. So we are stuck with a string split.

```go
const s3FolderSep = "/"

func extractBucketAndObject(s string) (string, string) {
  pathSplit := strings.split(s, s3FolderSep)
  bucket := pathSplit[0]
  object := strings.Join(pathSplit[1:], s3FolderSep)
  return bucket, object
}
```

Reply:
Oh!!! The simple ones get you. I realize what the issue was. Apologies for that confusion. Tested it with folders inside a bucket.